udata/forms/fields.py

```diff
@@ -391,16 +408,29 @@ def __init__(self, model_form, *args, **kwargs):
         self.nested_model = model_form.model_class
         self.data_submitted = False
         self.initial_data = []
+        self.prefix = '{0}-'.format(self.name)
 
     def process(self, formdata, data=unset_value):
+        self._formdata = formdata
         self.initial_data = data
-        prefix = '{0}-'.format(self.name)
-        if formdata and any(k.startswith(prefix) for k in formdata):
+        self.has_data = formdata and any(k.startswith(self.prefix) for k in formdata)
```

Comment:
Yep, but I didn't want to factorize yet. It seems overkill to me for 2 attributes.
It's to becoming complex :/